src/Compilers/CSharp/Test/Emit2/Semantics/PatternMatchingTests5.cs

```diff
@@ -2613,5 +2613,503 @@ public interface INumberBase<T> where T : INumberBase<T>
+                    );
+            }
+        }
+
+        [Theory, CombinatorialData]
+        public void ShortestPathToDefaultNodeYieldsNoRemainingValues_Nullability_NullableBool(bool nullableEnable)
+        {
+            var source = """
+public enum Enum
+{
+    Zero = 0,
+    One = 1,
+    Two = 2,
+}
+
+class N
+{
+    private static int M1(Enum e1, Enum e2, bool? i)
+    {
+        return (e1, e2, i) switch // 1
+        {
+            (Enum.Two, _, _) => 0,
+            (_, Enum.Two, _) => 0,
+            (Enum.Zero, Enum.Zero, _) => 0,
+            (Enum.Zero, Enum.One, _) => 0,
+            (Enum.One, Enum.Zero, _) => 0,
+            ( < 0 or > Enum.Two, _, _) => 0,
+            (_, < 0 or > Enum.Two, _) => 0,
+            (_, _, true) => 1,
+        };
+    }
+
+    private static int M2(Enum e1, Enum e2, bool? i)
+    {
+        return (e1, e2, i) switch // 2
+        {
+            (Enum.Two, _, _) => 0,
+            (_, Enum.Two, _) => 0,
+            (Enum.Zero, Enum.Zero, _) => 0,
+            (Enum.Zero, Enum.One, _) => 0,
+            (Enum.One, Enum.Zero, _) => 0,
+            ( < 0 or > Enum.Two, _, _) => 0,
+            (_, < 0 or > Enum.Two, _) => 0,
+            (_, _, true) => 1,
+            (_, _, false) => 1,
+        };
+    }
+}
+""";
+            var comp = CreateCompilation(source, options: TestOptions.DebugDll.WithNullableContextOptions(nullableEnable ? NullableContextOptions.Enable : NullableContextOptions.Disable));
+            if (nullableEnable)
+            {
+                comp.VerifyDiagnostics(
+                    // (12,28): warning CS8509: The switch expression does not handle all possible values of its input type (it is not exhaustive). For example, the pattern '(Enum.One, Enum.One, false)' is not covered.
+                    //         return (e1, e2, i) switch // 1
+                    Diagnostic(ErrorCode.WRN_SwitchExpressionNotExhaustive, "switch").WithArguments("(Enum.One, Enum.One, false)").WithLocation(12, 28),
+                    // (27,28): warning CS8655: The switch expression does not handle some null inputs (it is not exhaustive). For example, the pattern '(Enum.One, Enum.One, null)' is not covered.
+                    //         return (e1, e2, i) switch // 2
+                    Diagnostic(ErrorCode.WRN_SwitchExpressionNotExhaustiveForNull, "switch").WithArguments("(Enum.One, Enum.One, null)").WithLocation(27, 28)
+                    );
+            }
+            else
+            {
+                comp.VerifyDiagnostics(
+                    // (12,28): warning CS8509: The switch expression does not handle all possible values of its input type (it is not exhaustive). For example, the pattern '(Enum.One, Enum.One, false)' is not covered.
+                    //         return (e1, e2, i) switch // 1
+                    Diagnostic(ErrorCode.WRN_SwitchExpressionNotExhaustive, "switch").WithArguments("(Enum.One, Enum.One, false)").WithLocation(12, 28)
+                    );
```

Comment:
>); [](http://example.com/codeflow?start=20&length=2)

I'm surprised we require `nullable enable` before requiring `(bool?)null` to be included in the `switch` expression for 2, but that is independent of this change.

Reply:
Presumably the reason is that nullable analysis (when enabled) could determine the case is not needed based on flow analysis.